src/meta/src/stream/source_manager.rs

```diff
@@ -1033,6 +1038,46 @@ impl SourceManager {
         for source_id in source_ids {
             if let Some(handle) = core.managed_sources.remove(&source_id) {
                 handle.handle.abort();
+                // remove the shared kafka client entry if it exists
+                if let Some(entry) = handle.share_client_entry {
+                    SHARED_KAFKA_CLIENT
+                        .entry_by_ref(&entry)
+                        .and_compute_with(|maybe_item| async {
+                            if let Some(item) = maybe_item {
+                                match item.into_value().as_ref() {
+                                    Ok(item_val) => {
+                                        let share_item = SharedKafkaItem {
+                                            client: item_val.client.clone(),
+                                            ref_count: item_val.ref_count - 1,
```

Comment:
I don't get why do we need `ref_count`. Can we just use sth like `get_with`?

Reply:
Then how do we manage the RdKafka client instance if all related sources are dropped? IIUC, if we remove the ref_count, the client instance will always be kept in memory, until restarting the meta node. There is no evict policy enabled for the cache.

Reply:
Oh... I'm a little confused now. Then should we use a concurrent hashmap like `dashmap`, instead of a cache..?

Reply:
As mentioned by Bugen, I think we can store `Weak` in the cache.

Reply:
Not sure if `Weak` fits well with `moka`. We may also try `dashmap` or `weak-table`.

Reply:
@BugenZhao Why do you think `Weak` might not work with `moka`? Actually I'm also thinking what's the difference between `moka` (cache) and `dashmap` (concurrent hashmap)

Reply:
> Actually I'm also thinking what's the difference between moka (cache) and dashmap (concurrent hashmap)

I think moka is a dashmap with evict policy and guarantees the updates can be done atomically.

Reply:
I think `dashmap` behaves similar to `moka::sync::Cache` with unlimited capacity. 🤔

> Why do you think `Weak` might not work with `moka`?

Because the interface does not seem to be _that_ compatible with storing a `Weak`, like, no auto-eviction, requires the caller to keep the strong reference:

```rust
 let mut client_arc: Option<Arc<KafkaClientType>> = None;
```

Reply:
> requires the caller to keep the strong reference

Isn't this expected usage? i.e., store `Weak` in the map, while store `Arc` in the caller. Otherwise who keeps the `Arc`? :eyes:

Reply:
I mean when inserting. 🤣 For example, `weak-table` allows you to pass a closure returning an `Arc` in `insert_with` while actually storing a `Weak` but returning an `Arc` back to the caller. With `moka` we need to temporarily hold the strong reference to prevent it from being deallocated.